openhtf/io/output/mfg_inspector.py

```diff
@@ -475,3 +492,24 @@ def __call__(self, test_record):  # pylint: disable=invalid-name
       filename = self._OutputToTestRunProto(test_record)
       logging.info('Saved local file: %s', filename)
       raise
+
+
+class UploadDirectory(object):
+  """Upload then delete test run pb files in a directory.
+
+  Args:
+    user: Google cloud service account for Oauth2client.
+    keydata: Google cloud key data for Oauth2client.
+    directory: Directory containing test run pb files.
+  """
+
+  def __init__(self, user, keydata, directory):
+    self._UploadToMfgInspector = UploadToMfgInspector(user, keydata)
+    self._directory = directory
+
+  def __call__(self, _):  # pylint: disable=invalid-name
+    filenames = glob.glob('%s*.pb' % self._directory)
+    for filename in filenames:
+      if self._UploadToMfgInspector.UploadTestRun(filename):
```

Comment:
do you want to try-except around this or only attempt to upload until the first one fails?

This can get really complicated, such as with things like delaying retries for each file, or delaying between attempts (circuit-breaker pattern-esque, so if you can't upload a file, stop trying for 5 minutes instead of retrying every time, such as if you have a quick 30-second test on a loop). I don't know if we want to provide those things or provide this and say anything more complicated should go outside until there's N+ users with it replicated? Maybe it should go in contrib/ like @jettisonjoe and I talked about before, so there's less expectation of "it should always work" and more of a "if it doesn't work, copy-paste it into your project and change it"?